test/hotspot/jtreg/testlibrary/ctw/Makefile

```diff
@@ -73,10 +74,20 @@ $(DST_DIR)/ctw.sh: $(DST_DIR)
 	echo '$${JAVA_HOME}/bin/java $${JAVA_OPTIONS} $(EXPORTS) -XX:-UseCounterDecay -Xbatch "-XX:CompileCommand=exclude,java/lang/invoke/MethodHandle.*" -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -Xbootclasspath/a:wb.jar -jar ctw.jar $$@' > $@
 	chmod a+x $@
 
+$(DST_DIR)/ctwrunner.sh: $(DST_DIR)
+	echo '$${JAVA_HOME}/bin/java $${JAVA_OPTIONS} -Dtest.jdk=$${JAVA_HOME} -jar ctwrunner.jar $$@' > $@
+	chmod a+x $@
+
 $(DST_DIR)/ctw.jar: filelist $(DST_DIR)/wb.jar
 	@mkdir -p $(OUTPUT_DIR)
 	$(JAVAC) $(EXPORTS) -sourcepath $(SRC_DIR) -d $(OUTPUT_DIR) -cp $(DST_DIR)/wb.jar @filelist
-	$(JAR) --create --file=$@ --main-class $(MAIN_CLASS) -C $(OUTPUT_DIR) .
+	$(JAR) --create --file=$@ --main-class $(CTW_MAIN_CLASS) -C $(OUTPUT_DIR) .
+	@rm -rf $(OUTPUT_DIR)
+
+$(DST_DIR)/ctwrunner.jar: filelist $(DST_DIR)/wb.jar
```

Comment:
we don't need ctwrunner.jar if we have a bash script. 
ctw.jar has already included CTWRunner.class.

Reply:
I think the jar file is needed. Jar files are associated with a main class. Not possible to `java -jar <specify main class>`, can only specify main class during `javac`.

Reply:
I know, but we don't have to launch CTWRunner using `-jar'. 

A jar file can contain a multiple of entry points. We can launch it in the conventional way. 
```
$java -cp ctw.jar $CTW_MAIN_CLASS
```